packages/bigquery/src/dataset.js

```diff
@@ -70,6 +78,11 @@ function Dataset(bigQuery, id) {
      *
      * @example
      * dataset.exists(function(err, exists) {});
+     *
+     * //-
+     * // If the callback is omitted, we'll return a Promise.
+     * //-
+     * dataset.exists().then(function(exists) {});
```

Comment:
I feel like we should choose consistency here, that way users always know what to expect and won't be surprised. Otherwise, I feel like one what constantly have to be checking the API ref docs for google-cloud-node to figure out which methods return arrays vs which ones don't.